### ISSUE TITLE
[substyle-jss] ProvideSheet is not compatible with react-jss v10

### ISSUE THREAD

**Issue by tzimmermann** [jfschwarz/substyle#51]
v10 of `react-jss` (released last September) removed the `inject` prop on their HoC:

https://github.com/cssinjs/jss/blob/master/changelog.md#breaking-changes-2

Since `substyle-jss` is relying on that one (see below) and I don't see an alternative way of getting the sheet, this could be a tough one. @jfschwarz any ideas how to handle this? 

https://github.com/jfschwarz/substyle/blob/8bc6710bf782fd0eff12413f5560a62ed7c9cb0a/packages/substyle-jss/src/ProvideSheet.js#L13

Context: after upgrading `react` in pex-client, a `react-jss` upgrade was necessary to get rid of deprecated lifecycle methods warning. But this is now blocked by this issue.

**Comment by tzimmermann** [comment]
Ah I just noted there is a PR https://github.com/jfschwarz/substyle/pull/44
Do you need any help with that?

**Comment by jfschwarz** [comment]
@tzimmermann Yes, I handled the JSS upgrade in the big breaking API PR. I guess it would make sense to extract the JSS upgrade into its own PR.

I also believe this upgrade could significantly improve your rendering performance, which I believe has been an issue in PI? The current substyle-jss adapter was really just a quick-and-dirty POC. In PR #44 I managed to come up with something more sensible, if I remember correctly. Since PI is the only user of this adapter it would make sense for you to take a bit of ownership of it. If you want we can schedule a little pair programming session for this.
